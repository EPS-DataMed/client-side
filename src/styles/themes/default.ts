export const defaultTheme = {
  colors: {
    neutral: '#FFFFFF',
    neutral50: '#F0F0F0',
    neutral100: '#F9F9F9',
    neutral200: '#E5E5E5',
    neutral300: '#CCCCCC',
    neutral400: '#A5A5A5',
    neutral500: '#7F7F7F',
    neutral600: '#42424A',
    neutral700: '#36363B',
    neutral900: '#181818',
    blue500: '#2980B9',
    blue440: '#34CDFF',
    green500: '#3fd255',
    red500: '#FF0000',
  },
  fontSizes: {
    xxs: '0.625rem',
    xs: '0.75rem',
    sm: '0.875rem',
    md: '1rem',
    lg: '1.125rem',
    xl: '1.25rem',
    '2xl': '1.5rem',
    '4xl': '2rem',
    '5xl': '2.25rem',
    '6xl': '3rem',
    '7xl': '4rem',
    '8xl': '4.5rem',
    '9xl': '6rem',
  },
  fontWeights: {
    regular: '400',
    medium: '500',
    bold: '700',
  },
  fonts: {
    primary: 'Open Sans, sans-serif',
    secondary: 'Poppins, sans-serif',
  },
  lineHeights: {
    shorter: '125%',
    short: '140%',
    base: '160%',
    tall: '180%',
  },
  radii: {
    px: '1px',
    xs: '4px',
    sm: '6px',
    md: '8px',
    lg: '16px',
    full: '99999px',
  },
  space: {
    1: '0.25rem',
    2: '0.5rem',
    3: '0.75rem',
    4: '1rem',
    5: '1.25rem',
    6: '1.5rem',
    7: '1.75rem',
    8: '2rem',
    9: '2.25rem',
    10: '2.5rem',
    12: '3rem',
    16: '4rem',
    19: '4.75rem',
    20: '5rem',
    27: '6.75rem',
    40: '10rem',
    64: '16rem',
    80: '20rem',
  },
} as const
