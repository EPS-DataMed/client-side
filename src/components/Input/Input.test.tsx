import { render } from '@testing-library/react'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import { Input } from './'
import { defaultTheme } from '../../styles/themes/default'

describe('Input Component', () => {
  it('should render the input component', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.Input />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should apply error styles when hasError is true', () => {
    const { getByPlaceholderText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.Input hasError={true} placeholder="Test input" />
      </ThemeProvider>,
    )
    const input = getByPlaceholderText('Test input')
    expect(input).toHaveStyleRule(
      'border',
      `1px solid ${defaultTheme.colors.red500}`,
    )
  })

  it('should apply searchbar styles when variant is searchbar', () => {
    const { getByPlaceholderText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.Input variant="searchbar" placeholder="Search" />
      </ThemeProvider>,
    )
    const input = getByPlaceholderText('Search')
    expect(input).toHaveStyleRule('height', '32px')
    expect(input).toHaveStyleRule('padding-left', '38px')
    expect(input).toHaveStyleRule('padding-right', '38px')
  })

  it('should render the label component', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.Label>Label Text</Input.Label>
      </ThemeProvider>,
    )
    expect(getByText('Label Text')).toBeInTheDocument()
  })

  it('should render the error message component', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.ErrorMessage>Error Message</Input.ErrorMessage>
      </ThemeProvider>,
    )
    expect(getByText('Error Message')).toBeInTheDocument()
  })

  it('should render the description component', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.Description>Description Text</Input.Description>
      </ThemeProvider>,
    )
    expect(getByText('Description Text')).toBeInTheDocument()
  })

  it('should apply error styles to RequiredText when variant is error', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.RequiredText variant="error">Required Text</Input.RequiredText>
      </ThemeProvider>,
    )
    expect(getByText('Required Text')).toHaveStyleRule(
      'color',
      defaultTheme.colors.red500,
    )
  })

  it('should render the error message root component', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Input.ErrorMessageRoot>
          <span>Error Icon</span>
          <Input.ErrorMessage>Error Message</Input.ErrorMessage>
        </Input.ErrorMessageRoot>
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
