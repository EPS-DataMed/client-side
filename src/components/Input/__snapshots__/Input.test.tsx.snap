// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render the error message root component 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c1 {
  color: #FF0000;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
}

<div
  class="c0"
>
  <span>
    Error Icon
  </span>
  <span
    class="c1"
  >
    Error Message
  </span>
</div>
`;

exports[`Input Component should render the input component 1`] = `
.c0 {
  height: 35px;
  background-color: inherit;
  width: 100%;
  border-radius: 8px;
  color: #181818;
  font-size: 0.875rem;
  font-weight: 400;
  box-sizing: border-box;
  border: 1px solid #42424A;
  padding: 0 1rem;
  transition: all 0.3s;
}

.c0:focus {
  border: 1px solid #2980B9;
}

.c0:hover {
  border: 1px solid #2980B9;
}

.c0::placeholder {
  color: #A5A5A5;
}

<input
  class="c0"
/>
`;
